www/include/options/oreon/modules/installForm.php

```diff
@@ -113,25 +113,45 @@
 $form1 = new HTML_QuickFormCustom('Form', 'post', "?p={$p}");
 
 if ($form1->validate()) {
+    $tpl->assign('p', $p);
+    $tpl->assign('o', $o);
+    $tpl->assign('name', $name);
+
     $moduleInstaller = $moduleFactory->newInstaller($name);
     $insert_ok = $moduleInstaller->install();
 
     if ($insert_ok) {
-        $tpl->assign('output', _('Module installed and registered'));
+        $tpl->assign('msg', 1);
 
         /* Rebuild modules in centreon object */
         $centreon->creatModuleList($pearDB);
         $centreon->user->access->updateTopologyStr();
         $centreon->initHooks();
     } else {
-        $tpl->assign('output', _('Unable to install module'));
+        $tpl->assign('msg', 2);
     }
+
+    $tpl->display('reload.ihtml');
+    return;
 } elseif ($o == 'i' && !$moduleInfoObj->getInstalledInformation($name)) {
     $form1->addElement('submit', 'install', _('Install Module'), array('class' => 'btc bt_success'));
     $redirect = $form1->addElement('hidden', 'o');
     $redirect->setValue('i');
 }
 
+if (array_key_exists('msg', $_GET) && $_GET['msg']) {
```

Comment:
isn't it a $_POST instead a $_GET ?

Reply:
nope this parameter is in URL query after the POST request the page will be redirected to new URL with GET request